WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -66,51 +66,54 @@ private SystemAwakeChecker(CoinJoinManager coinJoinManager, Func<Task<IPowerSavi
 
 		protected override async Task ActionAsync(CancellationToken cancel)
 		{
-			IPowerSavingInhibitorTask? task = _powerSavingTask;
-
 			switch (CoinJoinManager.HighestCoinJoinClientState)
 			{
 				case CoinJoinClientState.Idle:
-					if (task is not null)
-					{
-						Logger.LogWarning("Computer idle state is allowed again.");
-						await task.StopAsync().ConfigureAwait(false);
-						_powerSavingTask = null;
-					}
+					await ReleaseAllPreventionAsync().ConfigureAwait(false);
+					break;
 
+				case CoinJoinClientState.InProgress or CoinJoinClientState.InCriticalPhase:
+					await PreventSleepAsync().ConfigureAwait(false);
 					break;
 
-				case CoinJoinClientState.InProgress:
-					if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-					{
-						if (task is not null)
-						{
-							if (!task.Prolong(Timeout.Add(TimeSpan.FromMinutes(1))))
-							{
-								Logger.LogTrace("Failed to prolong the power saving task.");
-								task = null;
-							}
-						}
-
-						if (task is null)
-						{
-							Logger.LogTrace("Create new power saving prevention task.");
-							_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
-						}
-					}
-					else
-					{
-						await EnvironmentHelpers.ProlongSystemAwakeAsync().ConfigureAwait(false);
-					}
+				default:
+					throw new ArgumentOutOfRangeException();
```

Comment:
I would switch `ArgumentOutOfRangeException` for

```suggestion
					throw new NotSupportedException($"Unsupported {CoinJoinManager.HighestCoinJoinClientState} value.");
```

Then we would clearly convey "One may add new enum values but we don't support them -> bug."